### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by BusyJay**
- mention async ready
- fix doc warnings

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -412,6 +418,15 @@ to advance the applied index inside.
 
 For more information, check out an [example](examples/single_mem_node/main.rs#L113-L179).
 
+Sometimes it's better not to block the raft machine in IO operation, so that latency of
+read/write can be more predictable and the fsync frequencey can be controlled. The crate
+supports async ready to offload the IO operation to other thread. The usage is the same as
+above except:
+1. all writes are not required to be persisted, they can be written into memory caches;
```
Comment (NingLin-P) [inline src/lib.rs:425]:
```suggestion
1. Before handling the next write, all previous writes are not required to be persisted, they can be written into memory caches;
```

File: src/lib.rs [hunk 2]
```diff
@@ -412,6 +418,15 @@ to advance the applied index inside.
...
+2. persisted messages should be sent after all coresponding writes are persisted;
+3. [`advance_append_async`](RawNode::advance_append_async) is used when all writes are finished
+    instead of `advance/advance_append`.
```
Comment (NingLin-P) [inline src/lib.rs:428]: Should also mention what function to call when the writes are persisted.

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM

File: src/lib.rs (hunk 1 above)
Comment (BusyJay) [reply src/lib.rs:425]: What's the difference?

File: src/lib.rs (hunk 2 above)
Comment (BusyJay) [reply src/lib.rs:428]: It's mentioned in `advance_append_async`.

File: src/lib.rs (hunk 1 above)
Comment (NingLin-P) [reply src/lib.rs:425]: `all writes are not required to be persisted` looks like no need to persist the write at all.

Comment (BusyJay) [reply src/lib.rs:425]: How about "all writes are not required to be persisted immediately"?

Comment (NingLin-P) [reply src/lib.rs:425]: lgtm

**Review by NingLin-P — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]
LGTM

**Pull request merged**